lib/AST/Decl.cpp

```diff
@@ -3717,6 +3732,21 @@ static bool checkAccessUsingAccessScopes(const DeclContext *useDC,
          VD->getDeclContext()->getParentModule() == useDC->getParentModule();
 }
 
+/// Checks if \p VD is an ObjC member implementation:
+///
+/// \li It's in an \c \@_objcImplementation extension
+/// \li It's \c \@objc
+/// \li Its access level is not \c private or \c fileprivate
+static bool
+isObjCMemberImplementation(const ValueDecl *VD,
```

Comment:
I don't love this name, or some of the similar ones.  "Implementation" is not a word that ambiguously refers to an ObjC `@implementation` declaration, and there are a lot of places where `ObjC` means e.g. the `@objc` attribute.  Is there a more distinctive term we can use within the compiler for these things?